### ISSUE TITLE
Expose `Data` struct and create document from an instance

### ISSUE THREAD

**Issue by hodlen** [Brendonovich/prisma-client-rust#153]
Currently, we have a `Data` struct for each model with serializing and deserializing capabilities. From a user's view, creating a new model from such an instance will be great. 

Is this feasible? As far as I know, it might need some extra effort for optional fields (like MongoDB's `id`) and some internal types (like DateTime). 

Please correct me if I misunderstood something. And I can help with the dev works.

**Comment by Brendonovich** [comment]
While this could be doable, it goes directly against why I designed Prisma Client Rust the way I did haha.

This sort of pattern is what ORMs like [SeaORM](https://github.com/SeaQL/sea-orm) do, and while it can be nice it requires either specifying every field to create or using `Default`, which is something I decided against [here](https://github.com/Brendonovich/prisma-client-rust/discussions/100). Instead I opted for the field-by-field creation model, with a `Vec` for all non-required fields.

To keep with this model, I could add a method on `Data` that calls `model::create` with the struct's data?

**Comment by hodlen** [comment]
I agree with your main point, the only suitable scenenario for this is about bulk creating. It is not a very common case. If it would introduce more trouble than its benefit and breaks the system's design, it might not be a good idea to implement.

The underlying problem is that I am wondering if there some good practice to decouple the creation of the entity creation and DB operations. Use `Data` struct as an intermediate data transfer object comes to my mind first, so I raised this issue. 

In this system, is there something to serve for this? I think the `Create` struct might also be capable for that?

**Comment by Brendonovich** [comment]
I'm not quite sure what you mean by 'the only suitable scenario for this is about bulk creating'.
The `model::create()` function allows you to decouple assembling a `Vec` of create data from a call to `create_many`, unless you're using something other than `create_many`?

**Comment by hodlen** [comment]
Sorry for the misunderstanding. I just meant creating in the former comment. 

Well as you said and shown in the `create_many` sample code, I guess using `Create` to decouple creating data entity and real IO is the preferred way in this system? 

> I'm not quite sure what you mean by 'the only suitable scenario for this is about bulk creating'.
> The `model::create()` function allows you to decouple assembling a `Vec` of create data from a call to `create_many`, unless you're using something other than `create_many`?

**Comment by Brendonovich** [comment]
Yeah pretty much, there isn't a `Create` struct but there is the `create` model function. In case you thought it returns a struct, it returns a tuple as documented [here](https://prisma.brendonovich.dev/writing-data/create#create-many).

**Comment by hodlen** [comment]
Thanks for the help! This is the solution for me. And also, you could close this issue if you do not like this idea.

**Comment by Brendonovich** [comment]
I'm gonna implement something like this with `Create` structs
![image](https://user-images.githubusercontent.com/14191578/235286763-eac183cb-bd30-468f-9b43-81e8a1288d67.png)
